src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -804,43 +807,40 @@ private int ReadBuffer(Span<char> userBuffer, out bool readToUserBuffer)
                 }
             }
 
-            StringBuilder? sb = null;
+            using ValueStringBuilder sb = new(0);
```

Comment:
This should start off using stackalloc'd space, e.g. `var sb = new ValueStringBuilder(stackalloc char[256]);`.

Reply:
Done